### ISSUE TITLE
Fix Request: false positive on UnconditionalCreatedLogArguments

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#704]
The rule should not violate on use of size() on java.util implementation types because this is typically very cheap and can usually be inlined by the compiler.

`LOGGER.debug("Created number of fluffs: {}", fluffs.size());`

The same holds for `isEmpty()`, `Map.containsKey(k)`(except TreeMap), `Set.contains(e)` (except TreeSet)

**Comment by jborgers** [comment]
Turns out the method (size()) is not the reason for the invalid match. It is because the argument is not functionally created yet with a side effect:

```
void good(String input) {
    List fluffs = createAndStoreFluffsFromInput(input);
    LOGGER.debug("Created number of fluffs: {}", fluffs.size()); // <-- false positive
}
```
Note that the List is only used for logging.

**Comment by jborgers** [comment]
Hard to know whether a method returning a log argument does not have side effects.
Examples:
```
arg = obj.toString(); // no side effects and may be expensive, a violation
arg = list.get(i); // no side effects but not expensive, no violation
arg = complexCalculation(); // likely no side effects, but not sure, it might e.g. modify fields

```

**Comment by jborgers** [comment]
As mentioned to me by @stokpop, I think it is unusual and typically bad practice to not use a returned value in a method. Only using it unconditionally in (trace/debug/info) logging is a reason for a violation. 

We could document that for argument creating methods with side effects, it might be a false positive.
